codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -351,20 +528,114 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId,
-
+            onHttp2Frame(ctx, new DefaultHttp2DataFrame(data, endOfStream, padding)
+                                        .stream(requireStream(streamId)).retain());
             // We return the bytes in consumeBytes() once the stream channel consumed the bytes.
             return 0;
         }
+
+        @Override
+        public void onGoAwayRead(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData) {
+            onHttp2Frame(ctx, new DefaultHttp2GoAwayFrame(lastStreamId, errorCode, debugData).retain());
+        }
+
+        @Override
+        public void onPriorityRead(
+                ChannelHandlerContext ctx, int streamId, int streamDependency, short weight, boolean exclusive) {
+            // TODO: Maybe handle me
+        }
+
+        @Override
+        public void onSettingsAckRead(ChannelHandlerContext ctx) {
+            // TODO: Maybe handle me
+        }
+
+        @Override
+        public void onPushPromiseRead(
+                ChannelHandlerContext ctx, int streamId, int promisedStreamId, Http2Headers headers, int padding)  {
+            // TODO: Maybe handle me
+        }
+
+        private Http2FrameStream requireStream(int streamId) {
+            Http2FrameStream stream = connection().stream(streamId).getProperty(streamKey);
+            if (stream == null) {
+                throw new IllegalStateException("Stream object required for identifier: " + streamId);
+            }
+            return stream;
+        }
+    }
+
+    void onUpgradeEvent(ChannelHandlerContext ctx, UpgradeEvent evt) {
+        ctx.fireUserEventTriggered(evt);
+    }
+
+    void onHttp2StreamWritabilityChanged(ChannelHandlerContext ctx, Http2FrameStream stream,
+                                         @SuppressWarnings("unused") boolean writable) {
+        ctx.fireUserEventTriggered(Http2FrameStreamEvent.writabilityChanged(stream));
+    }
+
+    void onHttp2StreamStateChanged(ChannelHandlerContext ctx, Http2FrameStream stream) {
+        ctx.fireUserEventTriggered(Http2FrameStreamEvent.stateChanged(stream));
+    }
+
+    void onHttp2Frame(ChannelHandlerContext ctx, Http2Frame frame) {
+        ctx.fireChannelRead(frame);
+    }
+
+    void onHttp2FrameStreamException(ChannelHandlerContext ctx, Http2FrameStreamException cause) {
+        ctx.fireExceptionCaught(cause);
+    }
+
+    private final class Http2RemoteFlowControllerListener implements Http2RemoteFlowController.Listener {
+        @Override
+        public void writabilityChanged(Http2Stream stream) {
+            Http2FrameStream frameStream = stream.getProperty(streamKey);
+            if (frameStream == null) {
+                return;
+            }
+            onHttp2StreamWritabilityChanged(
+                    ctx, frameStream, connection().remote().flowController().isWritable(stream));
+        }
+    }
+
+    /**
+     * {@link Http2FrameStream} implementation.
+     */
+    // TODO(buchgr): Merge Http2FrameStream and Http2Stream.
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)